### PR TITLE
chapter_10:Updating, showing and deleting users

### PR DESCRIPTION
**Pull request by ngyihkai**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,7 +1,14 @@
 class UsersController < ApplicationController
-  before_action :load_user, only: :show
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb:2]: why you remove this line?
You should keep it, and not change `show` method
```
def show; end
```

File: app/models/user.rb [hunk 2]
```diff
@@ -20,6 +21,10 @@ def forget
     update_attribute :remember_digest, nil
   end
 
+  def edit
```
Comment (huongnt-2545) [inline app/models/user.rb]: remove this method from here

File: app/views/shared/_error_messages_html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<% if object.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      <%= t (:the_form_contains)%>
+      <%= pluralize @user.errors.count, t(:error) %>.
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
```
Comment (huongnt-2545) [inline app/views/shared/_error_messages_html.erb]: indent
```suggestion
      <% @user.errors.full_messages.each do |msg| %>
        <li><%= msg %></li>
      <% end %>
```

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<% provide(:title, t(:all_users)) %>
+<h1><%= t(:all_users)%></h1>
+
+<%= will_paginate @users %>
+
+<ul class="users">
+  <% @users.each do |user| %>
+    <%= render @users %>
```
Comment (huongnt-2545) [inline app/views/users/index.html.erb]:
```suggestion
    <%= render user %>
```

File: db/seeds.rb [hunk 5]
```diff
@@ -5,3 +5,23 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+User.create!(name: "Example User",
+             email:"example@railstutorial.org",
+             password:              "111111",
```
Comment (huongnt-2545) [inline db/seeds.rb]:
```suggestion
             password: "111111",
```

File: db/seeds.rb [hunk 6]
```diff
@@ -5,3 +5,23 @@
...
+             password_confirmation: "111111")
+
+User.create!(name: "Example User",
+             email:"welcome@railstutorial.org",
+             password:              "111111",
```
Comment (huongnt-2545) [inline db/seeds.rb]:
```suggestion
             password: "111111",
```

**Comment by ngyihkai** [comment]
Ready

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -1,5 +1,11 @@
...
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb:2]: please keep this line, DON'T remove

File: app/views/users/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<%= form_for(@user) do |f| %>
+ <%= render "shared/error_messages", object: @user %>
```
Comment (huongnt-2545) [inline app/views/users/_form.html.erb]: indentation = 2 spaces

File: app/views/users/_user.html.erb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<li>
+  <%= gravatar_for user, size: 50 %>
+  <%= link_to user.name, user %>
+  <% if current_user.admin? && !current_user?(user) %>
+ | <%= link_to t(:delete), user, method: :delete, data: {confirm: t(:You_sure?)} %>
```
Comment (huongnt-2545) [inline app/views/users/_user.html.erb]:
```suggestion
   | <%= link_to t(:delete), user, method: :delete, data: {confirm: t(:You_sure?)} %>
```

**Comment by ngyihkai** [comment]
Ready

**Review by huongnt-2545 — COMMENTED** [review]
Still comments are not fixed yet
please resolve them

**Comment by ngyihkai** [comment]
Ready

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -1,6 +1,14 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
```
Comment (ghost) [inline app/controllers/users_controller.rb]:
```suggestion
  before_action :logged_in_user, except: %i(new show create)
```

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -1,6 +1,14 @@
...
+  before_action :correct_user, only: [:edit, :update]
```
Comment (ghost) [inline app/controllers/users_controller.rb]:
```suggestion
  before_action :correct_user, only: %i(edit update)
```

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ngyihkai** [comment]
Ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**